scripts/l1_node.sh

```diff
@@ -75,12 +84,31 @@ if [ "x${STATE_SYNC_RPC1}" != "x" ]; then
     dasel put statesync.trust_hash -t string -v $TRUST_HEIGHT_HASH -f ${APP_HOME}/config/config.toml
 fi
 
-echo "Starting validator node"
-allorad \
-    --home=${APP_HOME} \
-    start \
-    --moniker=${MONIKER} \
-    --minimum-gas-prices=0${DENOM} \
-    --rpc.laddr=tcp://0.0.0.0:26657 \
-    --p2p.seeds=$SEEDS \
-    --p2p.persistent_peers $PEERS
+if [ "$UPGRADE" == "true" ]; then
+    if [ ! -d "/data/cosmovisor" ]; then
+        echo "initialize cosmovisor"
+        cp -R /cosmovisor /data/
+        cosmovisor init /data/cosmovisor/genesis/bin/allorad
+    fi
+
+    echo "Starting validator node with cosmovisor"
+    cosmovisor \
+        run \
+        --home=${APP_HOME} \
+        start \
+        --moniker=${MONIKER} \
+        --minimum-gas-prices=0${DENOM} \
+        --rpc.laddr=tcp://0.0.0.0:26657 \
+        --p2p.seeds=$SEEDS \
+        --p2p.persistent_peers $PEERS
+else
+    echo "Starting validator node without cosmovisor"
+    allorad \
```

Comment:
Why it first runs with cosmovisor, then without?

Reply:
this is an if statement. if UPGRADE is true, then run with cosmovisor, else, run without as a normal binary

Reply:
sorry! reviewed from mobile, I may have missed the else lol